components/constellation/constellation.rs

```diff
@@ -599,17 +599,24 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
 
         let (event_loop, host) = match sandbox {
             IFrameSandboxState::IFrameSandboxed => (None, None),
-            IFrameSandboxState::IFrameUnsandboxed => match reg_host(&load_data.url) {
-                None => (None, None),
-                Some(host) => {
-                    let event_loop = self.event_loops.get(&top_level_frame_id)
-                        .and_then(|map| map.get(&host))
-                        .and_then(|weak| weak.upgrade());
-                    match event_loop {
-                        None => (None, Some(host)),
-                        Some(event_loop) => (Some(event_loop.clone()), None),
-                    }
-                },
+            IFrameSandboxState::IFrameUnsandboxed => {
+                match load_data.creator_pipeline_id.and_then(|cpi| self.pipelines.get(&cpi)) {
+                    Some(creator_pipeline) => {
+                        (Some(creator_pipeline.event_loop.clone()), None)
```

Comment:
We shouldn't always use the creator event loop, if there is a parent shouldn't we use its event loop?